### PR TITLE
don't block parallel execution of neighboring directories

### PR DESCRIPTION
**Pull request by jbinary**
for instance, if we have:
```
00_step1
  00_substep1
  05_substep2
00_step2
  00_substep1
  05_substep2
```

if 00_step2.00_substep1 is done before 00_step1.00_substep1, start 00_step2.05_substep2 right away as it has nothing to do with 00_step1 and shouldn't be blocked by it

### CODE REVIEW

**Comment by jbinary** [comment]
I test it by replacing `(status, out) = ExecUtils.RunCmd(task_cmd, timeout, pipe_output, task_vars)` in `zeus/runner.py` with:
```
   import random                                                                                  
   time.sleep(random.random() * 3)                                                                
   status = 0                                                                                     
   out = 'OK'
```
then it does not really start the tasks but rather just sleeps for a random amount of time.

this leads `zeus run pipeline/vtap/process/run/*` to take about 10 seconds instead of 20s on vanilla zeus.

obviously, the benefit is really dependant on specific situation and we need to test in real life. zen internal should be a good place to start.

**Comment by jbinary** [comment]
it's a little bit messy because I had to fit it into currently horrible zeus' code. should be good enough to see if we are interested in this improvement tho

File: pylib/base/exec_utils.py [hunk 1]
```diff
@@ -72,24 +74,34 @@ def ExecuteParallel(args, pool_size=0, callback=PicklableCallback()):
...
+
+    for _ in range(pool_size):
+      Thread(target=worker, args=(in_queue, out_queue)).start()
+
     try:
-      res = pool.map(callback, args)
-      pool.terminate()
-      sys.stdout.flush()
-      return res
+      while True:
```
Comment (Ideonvn) [inline pylib/base/exec_utils.py:129]: Would this not lock the process when sigterm is raised?

Comment (jbinary) [reply pylib/base/exec_utils.py:129]: I could think that using a daemon thread could be useful but it just works in practice, I stopped it early so many times while testing, no issues around this

**Review by Ideonvn — APPROVED** [review]
Locally looks fine
@jbinary how do you want to test this?

**Comment by jbinary** [comment]
I can make a branch in `zenysis` with updated `requirements.txt` and we can run a zen internal pipeline from that branch? It's dockerized right so shouldn't affect others?

**Comment by Ideonvn** [comment]
> I can make a branch in `zenysis` with updated `requirements.txt` and we can run a zen internal pipeline from that branch? It's dockerized right so shouldn't affect others?

Yeah what I am thinking, sounds good. Lets give it a test.

**Pull request merged**